zipline/pipeline/factors/factor.py

```diff
@@ -1227,6 +1227,54 @@ def compute(self, today, assets, out, data):
     '''
     dtype = float64_dtype
 
+    def __getattr__(self, name):
+        if name in self.outputs:
+            return RecarrayFactor(factor=self, attribute=name)
+        else:
+            raise AttributeError(
+                "This factor has no output called '{}'.".format(name)
+            )
+
+    def __iter__(self):
+        if not self.outputs:
+            raise ValueError('This factor does not have multiple outputs.')
+        RecarrayFactor_ = partial(RecarrayFactor, self)
+        return iter(map(RecarrayFactor_, self.outputs))
+
+
+class RecarrayFactor(Factor):
+
+    siblings = {}
```

Comment:
This dict is going to be shared by all instances of `RecarrayFactor`.  You probably want to just have this be `NotSpecified` and make a new dict.